src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/AddCarry.PlatformNotSupported.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+using System;
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.Intrinsics.X86
+{
+    /// <summary>
+    /// This class provides access to Intel ADDCARRY hardware instructions via intrinsics
+    /// </summary>
+    [CLSCompliant(false)]
+    public abstract class AddCarry : Sse42
```

Comment:
`AddWithCarry` is not behind a CPUID flag that requires `SSE42`, so the class should not inherit from `Sse42`.

`AddWithCarry` is considered a "baseline" instruction (that is, it is always available, and has been since the original 8086). So, it should be exposed in a class called `X86Base` that inherits from nothing (we determined just `Base` wasn't suitable when reviewing the corresponding ARM class name).